src/Hyperbolic.jl

```diff
@@ -0,0 +1,240 @@
+        )
+    end
+    return nothing
+end
+
+distance(M::Hyperbolic, x, y) = acosh(max(-minkowski_dot(x, y), 1.))
+@doc doc"""
+    exp!(M, y, x, v[, t=1.0])
+
+computes the exponential map on the [`Hyperbolic`](@ref) space $\mathbb H^n$ eminating from `x`
+towards `v`, which is optionally scaled by `t`. The result is stored in `y`. The formula reads
+
+````math
+\exp_x v = \cosh(\sqrt{\langle v,v\rangle_{\mathrm{M}}})x
++ \sinh(\sqrt{\langle v,v\rangle_{\mathrm{M}}})\frac{v}{\sqrt{\langle v,v\rangle_{\mathrm{M}}}},
+````
+where $\langle\cdot,\cdot\rangle_{\mathrm{M}}$ denotes the [`minkowski_dot`](@ref).
+"""
+function exp!(M::Hyperbolic, y, x, v)
+    vn = sqrt(max(minkowski_dot(v, v),0.))
+    if vn < eps(eltype(x))
+        y .= x
+        return y
+    end
+    y .= cosh(vn)*x + sinh(vn)/vn*v
+    return y
+end
+
+function flat!(M::Hyperbolic, v::FVector{CotangentSpaceType}, x, w::FVector{TangentSpaceType})
+    copyto!(v.data, w.data)
+    return v
+end
+
+injectivity_radius(H::Hyperbolic, args...) = Inf
+
+@doc doc"""
+    inner(M, x, v, w)
+
+compute the Riemannian inner product for two tangent vectors `v` and `w`
+from $T_x\mathbb H^n$ of the [`Hyperbolic`](@ref) space $\mathbb H^n$ given by
+$\langle w, v \rangle_{\mathrm{M}}$ the [`minkowski_dot`](@ref) Minkowski
+inner product on $\mathbb R^{n+1}$.
+"""
+@inline inner(M::Hyperbolic, x, w, v) = minkowski_dot(w, v)
+
+@doc doc"""
+    log!(M, v, x, y)
+
+computes the logarithmic map on the [`Hyperbolic`](@ref) space $\mathbb H^n$, i.e., `v`
+corresponds to the tangent vector representing the [`geodesic`](@ref) starting from `x`
+reaches `y` after time 1 on the [`Hyperbolic`](@ref) space `M`.
+The formula reads for $x\neq y$
+
+```math
+\log_x y = d_{\mathbb H^n}(x,y)
+\frac{y-\langle x,y\rangle_{\mathrm{M}} x}{\lVert y-\langle x,y\rangle_{\mathrm{M}} x \rVert_2}
+```
+and is zero otherwise.
+"""
+function log!(M::Hyperbolic, v, x, y)
+    scp = minkowski_dot(x, y)
+    w = y + scp*x
+    wn = sqrt(max( scp.^2-1, 0.))
+    if wn < eps(eltype(x))
+        zero_tangent_vector!(M,v,x)
+        return v
+    end
+    v .= acosh(max(1.,-scp))/wn .* w
+    return v
+end
+
+@doc doc"""
+    minkowski_dot(a,b)
+computes the Minkowski inner product of two Vectors `a` and `b` of same length
+`n+1`, i.e.
+
+````math
+\langle a,b\rangle_{\mathrm{M}} = -a_{n+1}b_{n+1} + \displaystyle\sum_{k=1}^n a_kb_k.
+````
+"""
+minkowski_dot(a::AbstractVector,b::AbstractVector) = -a[end]*b[end] + sum( a[1:end-1].*b[1:end-1] )
+
+@doc doc"""
+    manifold_dimension(H::Hyperbolic)
+
+Return the dimension of the hyperbolic space manifold $\mathbb H^n$, i.e. $n$.
+"""
+manifold_dimension(::Hyperbolic{N}) where {N} = N
+
+"""
+    mean(
+        M::Hyperbolic,
+        x::AbstractVector,
+        [w::AbstractWeights,]
+        method = CyclicProximalPointEstimationMethod();
```

Comment:
This line does not apply to this method.

Reply:
I don't follow you? The default we use on hyperbolic is the newly implemented mean using the CPP?

Reply:
This is a bit confusing because the method of `mean!` below does not have the `method` argument. After looking more closely which method calls which one it should have the intended behaviour though.

Reply:
hm? I still don't follow you, the `statistics` has exactely this method and this line introduces a default if you don't provide a method (which is the here newly introduces CPP for mean)...

Reply:
To be precise, the one thing I _do_ overwrite (and hence the documentation) is the _default_ method for estimation. It is changed form the grad desc to the CPP. and that's what's documented.

Reply:
Yes, I understand. I was just confused but it's OK now.